components/script/dom/vrframedata.rs

```diff
@@ -51,10 +51,14 @@ impl VRFrameData {
         };
 
         unsafe {
-            framedata.left_proj.set(slice_to_array_buffer_view(global.get_cx(), &matrix));
-            framedata.left_view.set(slice_to_array_buffer_view(global.get_cx(), &matrix));
-            framedata.right_proj.set(slice_to_array_buffer_view(global.get_cx(), &matrix));
-            framedata.right_view.set(slice_to_array_buffer_view(global.get_cx(), &matrix));
+            let _ = Float32Array::create(global.get_cx(), matrix.len() as u32, Some(&matrix),
+                                         framedata.left_proj.handle_mut());
+            let _ = Float32Array::create(global.get_cx(), matrix.len() as u32, Some(&matrix),
+                                         framedata.left_view.handle_mut());
+            let _ = Float32Array::create(global.get_cx(), matrix.len() as u32, Some(&matrix),
+                                         framedata.right_proj.handle_mut());
+            let _ = Float32Array::create(global.get_cx(), matrix.len() as u32, Some(&matrix),
+                                         framedata.right_view.handle_mut());
```

Comment:
This is a GC hazard right now - the GC does not know about `framedata` yet, so if any of the `Float32Array::create` calls trigger a GC, the existing typed arrays will be GCed and `framedata` will store invalid pointers. We should call `reflect_dom_object` first before creating the arrays.